model/provision.go

```diff
@@ -0,0 +1,101 @@
+package model
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+// prepare DB
+func SetupDB(dsn string) {
+
+	_, pw, host, port, db := ParseDSN(dsn)
+
+	{
+		db, err := gorm.Open(postgres.Open(CreatDSN("postgres", pw, host, port, "postgres")), &gorm.Config{})
+		if err != nil {
+			log.Print(err)
+		}
+
+		// should replace creatdbuser.sh
+		execSQL(db, []string{
```

Comment:
Would we be able to use this script to setup a new AWS RDS instance?  Guessing this bit with the hardcoded user / password might need tweaking

Reply:
There is an existing program which does that https://github.com/ONSdigital/dp-find-insights-poc-api/blob/develop/cmd/creatschema/main.go 

Some of that code is refactored into this package. Yes the password shouldn't be hardcoded (it's a dev one) and it needs further work to provision on prod. I'll add a comment to say that.